lib/kubernetes-deploy/runner.rb

```diff
@@ -19,11 +19,11 @@ def self.with_friendly_errors
       exit 1
     end
 
-    def initialize(namespace:, environment:, current_sha:, template_folder: nil, context:)
+    def initialize(namespace:, environment:, current_sha:, context:, template_dir: nil)
       @namespace = namespace
       @context = context
       @current_sha = current_sha
-      @template_path = File.expand_path('./' + (template_folder || "config/deploy/#{environment}"))
+      @template_dir = template_dir || File.expand_path("./config/deploy/#{environment}")
```

Comment:
Not sure where you were looking, but this variable drives the `template_dir` option: https://github.com/Shopify/shipit-engine/blob/master/app/models/shipit/deploy_spec/kubernetes_discovery.rb#L14. I.e. if you have this in your shipit.yml:

```yaml
kubernetes:
  namespace: my-app
  context: tier4
  template_dir: config/deploy/k8s/
```
This PR would make it so you'd need to know exactly where on your Shipit machine your repo gets cloned, and if that changes, it'll need to be updated on N repos. Note that the env var can also be set directly via the `machine.environment` section.

That said, this behaviour is kinda an artifact of the pre-first-class api. I'd be fine with making  `kubernetes.template_dir` result in an argument to the script (still a relative path) instead of the setting of an env var, and changing this env var to be an override intended for local development. But that's not how it works right now.